Create Big Sorting.cpp

srikanth7785 opened the pull request:
A must include problem under `Big Integers` 🤞

srikanth7785 committed:
Create Big Sorting.cpp

A must include problem under `Big Integers` 🤞

srikanth7785 committed:
Update README.md

srikanth7785 committed:
Update Big Sorting.cpp

Added Problem Statement

srikanth7785 reviewed (commented):
all done..

ajitzero commented on Big Integers/Big Sorting.cpp:
```diff
@@ -0,0 +1,42 @@
+/*
```
Please follow the same code format as the other programs. Use [this program](https://github.com/AjitZero/algo-cpp/blob/master/Dynamic%20Programming/LongestCommonSubsequence.cpp) as reference.

ajitzero replied:
```
/**
 * 
 * Title: <Specific title>
 * Description: <Specific description>
 * Author: <Your name>
 * GitHub: <Your link>
 * Twitter: <Your link (optional)>
 * 
 */
```

ajitzero commented on Big Integers/Big Sorting.cpp:
```diff
@@ -0,0 +1,42 @@
+using namespace std;
+bool customSorting(string a, string b){
+    if(a.length() == b.length()) // if both the strings are of same length
+        return a < b; // check if first string is less than the second string
```
Refactor to use [`std::string::compare`](https://www.geeksforgeeks.org/stdstringcompare-in-c/) for type safety.

ajitzero replied:
📌 Skipping this for now.

ajitzero commented on Big Integers/Big Sorting.cpp:
```diff
@@ -0,0 +1,42 @@
+    for(int i = 0;i < n; i++)
+    {string a; cin >> a; s.push_back(a);}
+    sort(s.begin(),s.end(),customSorting); // this is an inbuilt method available in `algorithm`. it will call `customSorting()` for each comparision
+    for(string i : s) //finally, print the output, which is sorted.
```
Use better variables names here, as we are treating it as a collection of strings. `i` is used exclusively for indices. Avoid single-letter variable names except for `N` (for length) and `i, j, k` for indices. Use `x, y, z` for lambda functions exclusively if possible or for coordinate systems.

ajitzero commented on Big Integers/Big Sorting.cpp:
```diff
@@ -0,0 +1,42 @@
+-------------------------
+Sample Input: 
+6
+31415926535897932384626433832795
```
Use more big numbers as an example to test large numbers of equal length to illustrate greater, lesser and equal values.

srikanth7785 replied:
On it..

ajitzero reviewed (changes requested):
Hi @srikanth7785! Thanks for your PR! I've reviewed your code and requested some changes. Please look into resolving those comments and [re-request a review](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review) for this PR.

Additionally, please add support for **handling negative numbers** or specifically mention a constraint for positive numbers in the problem description.

AjitZero commented:
Also, please rename the file as per the existing format to not have spaces. You can simply remove the space or use an underscore (`_`).

srikanth7785 commented:
Sure..will make necessary changes..

srikanth7785 committed:
Made necessary changes

* Added `Title`, `Description`, etc..,
* Improved variable names.
* handled `-ve` numbers case
* improved sample input and output, as requested.

srikanth7785 committed:
Update README.md

srikanth7785 commented:
- [x] Handling negative numbers.
- [x] Added `Title`, `Description`, `Author` etc..,
- [x] Use better variable names.
- [x] Added better Sample Input & Sample Output.
- [x] Rename file as per existing format.
- [ ] Refactor to use `std::string::compare` for type safety.

ajitzero commented on Big Integers/BigSorting.cpp:
```diff
@@ -0,0 +1,78 @@
+}
+/*
+-------------------------
+Sample Input:                                                                                   Sample Output:
```
Please move the sample output to _after_ the input, not side-by-side. It's more difficult to copy it for testing. Use the same coding style as previously used in other programs.

srikanth7785 replied:
I kept them side by side so that it will be easy to understand & compare input and output.

May be, it's a bit tough to copy for testing..but, still, that's only once right..or you can just give an input on your own, right..!

Anyway, I'll change it if you want me to..

Will be waiting for your reply..🤞⏳

ajitzero replied:
Please change it. It's also hard to review since I need a wide window to view this. The program width shouldn't exceed 80 characters as an industry standard, though I'm not following that strictly.

srikanth7785 replied:
fine..i'll do that..now..!

ajitzero commented on Big Integers/BigSorting.cpp:
```diff
@@ -0,0 +1,78 @@
+/*
+* Title: Big Sorting
+* Description: Given a list of numbers, which would even include numbers of some 20 digit numbers or so. And your task is to sort them.
```
Update this to:
```
Given a list of numbers, including Big Integers of an arbitrary number of digits, sort it in ascending order.
```

ajitzero reviewed (changes requested):
Let's leave the `std::string::compare` for now. Please fix the last few details & I'll reformat it before merging it.

srikanth7785 committed:
Update BigSorting.cpp

✔️ Updated Description.
✔️ Updated Sample Input & Sample Output.

ajitzero committed:
Reformatted the code

AjitZero commented:
@srikanth7785 Merged the PR! 🚀 Thank you for your first contribution to this repo.
